### PR TITLE
libc: Change errno to set_errno and get_errno

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

follow other code inside libc, please ignore:
```
Warning: /home/runner/work/nuttx/nuttx/nuttx/libs/libc/locale/lib_iconv.c:110:3: warning: #include outside of 'Included Files' 
```

## Impact

code refactor only

## Testing

ci

### CODE REVIEW

File: libs/libc/misc/lib_glob.c [hunk 1]
```diff
@@ -247,7 +247,8 @@ static int do_glob(FAR char *buf, size_t pos, int type, FAR char *pat,
 
       if (!type && lstat(buf, &st))
         {
-          if (errno != ENOENT && (errfunc(buf, errno) || (flags & GLOB_ERR)))
+          if (get_errno() != ENOENT &&
+              (errfunc(buf, get_errno() || (flags & GLOB_ERR)))
```
Comment (pkarashchenko) [inline libs/libc/misc/lib_glob.c:251]:
```suggestion
              (errfunc(buf, get_errno() || (flags & GLOB_ERR) != 0))
```

Comment (xiaoxiang781216) [reply libs/libc/misc/lib_glob.c]: Done.

File: libs/libc/misc/lib_glob.c [hunk 2]
```diff
@@ -247,7 +247,8 @@ static int do_glob(FAR char *buf, size_t pos, int type, FAR char *pat,
 
       if (!type && lstat(buf, &st))
         {
-          if (errno != ENOENT && (errfunc(buf, errno) || (flags & GLOB_ERR)))
+          if (get_errno() != ENOENT &&
+              (errfunc(buf, get_errno()) || (flags & GLOB_ERR)) != 0)
```
Comment (pkarashchenko) [inline libs/libc/misc/lib_glob.c:251]:
```suggestion
              (errfunc(buf, get_errno()) || (flags & GLOB_ERR) != 0))
```

File: libs/libc/misc/lib_glob.c [hunk 3]
```diff
@@ -287,16 +288,16 @@ static int do_glob(FAR char *buf, size_t pos, int type, FAR char *pat,
   dir = opendir(pos ? buf : ".");
   if (!dir)
     {
-      if (errfunc(buf, errno) || (flags & GLOB_ERR))
+      if (errfunc(buf, get_errno()) || (flags & GLOB_ERR))
```
Comment (pkarashchenko) [inline libs/libc/misc/lib_glob.c:291]:
```suggestion
      if (errfunc(buf, get_errno()) || (flags & GLOB_ERR) != 0)
```

File: libs/libc/misc/lib_glob.c [hunk 4]
```diff
@@ -344,19 +345,19 @@ static int do_glob(FAR char *buf, size_t pos, int type, FAR char *pat,
...
-  readerr = errno;
+  readerr = get_errno();
   if (p2)
     {
       *p2 = saved_sep;
     }
 
   closedir(dir);
-  if (readerr && (errfunc(buf, errno) || (flags & GLOB_ERR)))
+  if (readerr && (errfunc(buf, get_errno()) || (flags & GLOB_ERR)))
```
Comment (pkarashchenko) [inline libs/libc/misc/lib_glob.c:355]:
```suggestion
  if (readerr && (errfunc(buf, get_errno()) || (flags & GLOB_ERR) != 0))
```

Comment (xiaoxiang781216) [reply libs/libc/misc/lib_glob.c]: Done

File: libs/libc/misc/lib_glob.c (hunk 3 above)
Comment (xiaoxiang781216) [reply libs/libc/misc/lib_glob.c]: Done

File: libs/libc/misc/lib_glob.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/misc/lib_glob.c]: Done

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 Why?

follow the usage in libc, please search errno and [set|get]_errno in libs/libc.

File: libs/libc/stdio/lib_remove.c [hunk 5]
```diff
@@ -56,7 +56,7 @@ int remove(FAR const char *path)
    */
 
   if (unlink(path) != 0  /* If it is indeed a directory...  */
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_remove.c:58]: Should it be
```suggestion
  if (unlink(path) != 0        /* If it is indeed a directory...  */
```
now?

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_remove.c]: Done.

File: libs/libc/stdio/lib_remove.c [hunk 6]
```diff
@@ -55,9 +55,9 @@ int remove(FAR const char *path)
    * more frequently the necessary action.
    */
 
-  if (unlink(path) != 0  /* If it is indeed a directory...  */
-      && (errno != EPERM /* ...try to remove it.  */
-          || rmdir(path) != 0))
+  if (unlink(path) != 0     || /* If it is indeed a directory...  */
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_remove.c:58]: why `&&` is changed to `||`?

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_remove.c]: Typo, fixed.

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@pkarashchenko please ignore the false ci warning.

**Pull request merged**